storagepb: rename to storageconfig, don't use protobuf

RaduBerinde opened the pull request:
#### storagepb: rename to storageconfig, don't use protobuf

There is no reason to use protos to define the storage configuration.
This change redefines the structures without using proto and renames
the package.

Epic: none
Release note: None

#### storageconf: rename constants

Rename constants that had protobuf-style names.

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/146888)
<!-- Reviewable:end -->

annrpom reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 41 of 41 files at r1, 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/146888#-:-OQdZWe975xS31wzdjuO:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @jbowens)



<!-- Sent from Reviewable.io -->

RaduBerinde committed:
storagepb: rename to storageconfig, don't use protobuf

There is no reason to use protos to define the storage configuration.
This change redefines the structures without using proto and renames
the package.

Epic: none
Release note: None

RaduBerinde committed:
storageconfig: rename constants

Rename constants that had protobuf-style names.

Epic: none
Release note: None

jbowens reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 30 of 41 files at r1, 11 of 11 files at r3, 5 of 5 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/146888#-:-OQjGyjb4x5suDunRMCG:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (and 1 stale) (waiting on @RaduBerinde)



<!-- Sent from Reviewable.io -->

RaduBerinde commented:
TFTRs!

bors r+
